lib/srv/desktop/windows_server.go

```diff
@@ -595,25 +598,42 @@ func (s *WindowsService) startServiceHeartbeat() error {
 // service itself is running.
 func (s *WindowsService) startStaticHostHeartbeats() error {
 	for _, host := range s.cfg.Heartbeat.StaticHosts {
-		if err := s.startStaticHostHeartbeat(host, false); err != nil {
+		if err := s.startStaticHostHeartbeat(host); err != nil {
 			return err
 		}
 	}
+	for _, host := range s.cfg.Heartbeat.ADHosts {
+		if err := s.startOldStaticHostHeartbeat(host, true); err != nil {
+			return trace.Wrap(err)
+		}
+	}
 	for _, host := range s.cfg.Heartbeat.NonADHosts {
-		if err := s.startStaticHostHeartbeat(host, true); err != nil {
+		if err := s.startOldStaticHostHeartbeat(host, false); err != nil {
 			return trace.Wrap(err)
 		}
 	}
 	return nil
 }
 
-func (s *WindowsService) startStaticHostHeartbeat(host utils.NetAddr, nonAD bool) error {
+func (s *WindowsService) startOldStaticHostHeartbeat(addr utils.NetAddr, ad bool) error {
+	name, err := s.nameForStaticHost(addr.String())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return s.startStaticHostHeartbeat(servicecfg.WindowsHost{
+		Name:    name,
+		Address: addr,
+		AD:      ad,
+	})
+}
+
+func (s *WindowsService) startStaticHostHeartbeat(host servicecfg.WindowsHost) error {
 	heartbeat, err := srv.NewHeartbeat(srv.HeartbeatConfig{
 		Context:         s.closeCtx,
 		Component:       teleport.ComponentWindowsDesktop,
 		Mode:            srv.HeartbeatModeWindowsDesktop,
 		Announcer:       s.cfg.AccessPoint,
-		GetServerInfo:   s.staticHostHeartbeatInfo(host, s.cfg.HostLabelsFn, nonAD),
+		GetServerInfo:   s.staticHostHeartbeatInfo(host, s.cfg.HostLabelsFn),
```

Comment:
This means that the `host_labels` section will apply to `hosts`, `non_ad_hosts`, and the new `static_hosts` - is this what we intend?

Reply:
I think so, having rules here can be helpful if you have a bunch of desktops and you have apply the same label for many or all of them